src/sys.rs

```diff
@@ -2,14 +2,18 @@
 
 pub use self::inner::*;
 
-#[cfg(target_os = "redox")]
+#[cfg(any(target_os = "redox", not(feature = "std")))]
 mod inner {
-    use std::fmt;
-    use std::cmp::Ordering;
-    use std::ops::{Add, Sub};
+    use core::fmt;
+    use core::cmp::Ordering;
+    use core::ops::{Add, Sub};
+
+    #[cfg(feature = "std")]
```

Comment:
This entire module is gated, right? Does each statement here need a `#[cfg]`? Is there a way we can cut down on the `#[cfg]` traffic?